pkg/plugins/backendplugin/grpcplugin/client.go

```diff
@@ -40,7 +43,20 @@ var pluginSet = map[int]goplugin.PluginSet{
 	},
 }
 
-func newClientConfig(executablePath string, args []string, env []string, skipHostEnvVars bool, logger log.Logger,
+type clientTracerProvider struct {
+	tracer tracing.Tracer
+	embedded.TracerProvider
```

Comment:
Need one of `noop.TracerProvider`, `embedded.TracerProvider`, or `trace.TracerProvider` according to https://pkg.go.dev/go.opentelemetry.io/otel/trace@v1.30.0#hdr-API_Implementations? I am confused with 
**The API is designed to help inform an instrumentation author about this non-standard API evolution. It requires them to choose a default behaviour for unimplemented interface methods. There are three behavior choices they can make:**

    Compilation failure
    Panic
    Default to another implementation
But one of the behaviour will only happen when the user updates?

Reply:
Hmm right. So with `tracing.TracingService` you can get the `TracerProvider` via  
https://github.com/grafana/grafana/blob/658c79e1d50d19b461da6528884372d1c01de6ad/pkg/infra/tracing/tracing.go#L112-L114. But to be able to use that you would have to pass around `*tracing.TracingService` instead of `tracing.Tracer` which is not optimal. 

Getting a stronger and stronger feeling that given the needed changes it might not be worth it. Lets discuss in the team and decide

Reply:
Another option would be to cast the `tracing.Tracer` to some other interface, but that doesn't feel great either

Reply:
Should we put this issue on hold? Until we figure out an alternative solution to this.

Reply:
Sorry for my confusion. I think we can continue with this approach 👍